pagopa/pn-frontend
# Fix: PN-2613 Hide payment related fields after choosing "no payment" on a new notification

## Short description
Hide creditorTaxId and noticeCode fields when the user selected "no payment" in step 1 of a new notification

## List of changes proposed in this pull request
- set yup validation schema and form fields for recipient depending on payment type
- fix bug not properly clearing the touched state of formik after deleting an attachment

## How to test
- choose "no payment" at step 1 of the "new notification" procedure. The creditorTaxId and noticeCode should not be available for any of the recipients in the following step.
- remove an attachment box after having specified its "name", then click on "add another document..". The field name should not display any error message.
